server/src/main/java/org/apache/uniffle/server/storage/LocalStorageManager.java

```diff
@@ -69,8 +71,8 @@ public class LocalStorageManager extends SingleStorageManager {
   private final List<LocalStorage> localStorages;
   private final List<String> storageBasePaths;
   private final LocalStorageChecker checker;
-  private List<LocalStorage> unCorruptedStorages = Lists.newArrayList();
-  private final Set<String> corruptedStorages = Sets.newConcurrentHashSet();
+
+  private final Map<PartitionUnionKey, LocalStorage> partitionsOfStorage;
```

Comment:
Maybe it will occupy many memory.

Reply:
From our dashboard, there are only a few thousand partitions running at the same time

Reply:
And if we use the nested concurrent hashmap, I'm worry about the problem of thread safe.

Reply:
So far, yes